### PR TITLE
Story5

### PR DESCRIPTION
**Pull request by ellie-r**

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -64,59 +64,91 @@ function resetGame() {
...
+        startTimer(44, document.querySelector('#timer p'));
+    }
     listenForCollisions();
     listenForWinning();
     grimReaperZone.addEventListener('mouseleave',moveReaper);
-}
\ No newline at end of file
+}
+
+let timer;
```
Comment (mporam) [inline script.js]: i would piut this at the top with the other global variables/constants

File: script.js [hunk 2]
```diff
@@ -64,59 +64,91 @@ function resetGame() {
...
+
+function startTimer(duration, element) {
+    timerRunning = true;
+    let timeLeft = duration;
+    timer = setInterval(() => {
+        let minutes = parseInt(timeLeft / 60, 10);
```
Comment (mporam) [inline script.js]: if you only get 45 seconds, why the minutes??

File: script.js [hunk 3]
```diff
@@ -64,59 +64,91 @@ function resetGame() {
...
+        let seconds = parseInt(timeLeft % 60, 10);
+
+        minutes = (minutes < 10) ? ("0" + minutes) : minutes;
+        seconds = (seconds < 10) ? ("0" + seconds) : seconds;
+
+        element.textContent = minutes + ":" + seconds;
+
+        if (--timeLeft < 0) {
```
Comment (mporam) [inline script.js]: generally its not good to manipulate a variable while doing an `if` statement. do the decrement first, then do the comparison

File: script.js [hunk 4]
```diff
@@ -64,59 +64,91 @@ function resetGame() {
...
+// win if you get to Exit door
 function listenForWinning() {
     winningDoor.addEventListener('mouseenter', win);
 }
 function moveReaper() {
     reaperMove.play();
 }
 
-//call main function to start game from here
+let timerRunning = false;
```
Comment (mporam) [inline script.js]: again, global variables together at the top

File: script.js [hunk 5]
```diff
@@ -64,59 +64,91 @@ function resetGame() {
...
     e.stopPropagation();
     winModal.style.display = "none";
+    document.querySelector('#timer p').textContent = '00:45';
     start.addEventListener('mouseleave', runGame);
 });
 
 loseModal.addEventListener('click', (e) => {
     e.stopPropagation();
     loseModal.style.display = "none";
+    document.querySelector('#timer p').textContent = '00:45';
```
Comment (mporam) [inline script.js]: duplicate line with the win modal, this could be in a function

**Review by mporam — CHANGES_REQUESTED** [review]

File: script.js (hunk 4 above)
Comment (norayna) [inline script.js]: All variables and constants should be defined (and initialised if possible) at the start of the file. That way you can be sure no funny business happens if you accidentally try to use them before they are defined/initialised

File: script.js [hunk 6]
```diff
@@ -64,59 +64,91 @@ function resetGame() {
...
     reaperMove.cancel();
     grimReaperZone.removeEventListener('mouseleave', moveReaper);
+
 }
 
 // code here to activate win modal displaying
 function win() {
     winModal.style.display = "block";
+    clearInterval(timer);
+    timerRunning = false;
```
Comment (norayna) [inline script.js]: These two lines are the same for `win()` and `lose()` - put them in `resetGame()`?

File: script.js [hunk 7]
```diff
@@ -64,59 +64,91 @@ function resetGame() {
...
     loseModal.style.display = "block";
+    clearInterval(timer);
+    timerRunning = false;
     resetGame();
 }
 
 winModal.addEventListener('click', (e) => {
     e.stopPropagation();
     winModal.style.display = "none";
+    document.querySelector('#timer p').textContent = '00:45';
```
Comment (norayna) [inline script.js]: Using `'#timer p'` as query is a little bit dependent on HTML - should probably just put an id on the exact element you need

Comment (norayna) [reply script.js]: This is in multiple places

File: script.js [hunk 8]
```diff
@@ -64,59 +64,91 @@ function resetGame() {
...
-//listen for collisions with obstacles
+// listen for collisions with obstacles
 function listenForCollisions() {
     collision.forEach(item => {
         item.addEventListener('mouseenter', lose);
     });
 }
 
-//win if u get to Exit door (no timers yet)
+// win if you get to Exit door
```
Comment (norayna) [inline script.js]: The names of these functions are pretty clear so comments not really needed. The name `collision` could be a bit more semantic - I wouldn't immediately know what a "collision" is from reading the code. I don't think you should change that at this point but I'd say remove the comments

**Review by norayna — CHANGES_REQUESTED** [review]

**Review by mporam — APPROVED** [review]

**Review by norayna — APPROVED** [review]

**Pull request merged**